amd/src/mod_hsuforum_save_draft.js

```diff
@@ -0,0 +1,128 @@
+define([], function() {
+    /**
+     * Makes AJAX call to write current text for forum/discussion/post to DB for restoration later on
+     * @param forumid int
+     * @param discussionid int
+     * @param postid int
+     * @param text string
+     * @param userid int
+     */
+    function autoSave(forumid, discussionid, postid, text, userid) {
+        if ((forumid != '' || discussionid != '') && text != '' && userid != '') {
+            $.ajax({
+                url: "save_post_draft.php",
+                method: "POST",
+                data: {
+                    forumid: forumid,
+                    discussionid: discussionid,
+                    postid: postid,
+                    text: text,
+                    userid: userid
+                },
+                dataType: "text",
+                success: function() {
+                    // Do nothing, write-only call
+                }
+            });
+        }
+    }
+
+    /**
+     * Makes AJAX call to restore draft if it exists
+     * @param forumid
+     * @param discussionid
+     * @param postid
+     * @param userid
+     */
+    function fetchAutoSave(forumid, discussionid, postid, userid) {
+        if ((forumid != '' || discussionid != '' || postid != '') && userid != '') {
+            $.ajax({
+                url: "get_post_draft.php",
+                method: "GET",
+                data: {
+                    forumid: forumid,
+                    discussionid: discussionid,
+                    postid: postid,
+                    userid: userid
+                },
+                dataType: "text",
+                success: function(data) {
+                    window.drafttext = data;
+                }
+            });
+        }
+    }
+
+    return {
+        init: function(forumid, discussionid, userid) {
+            $(function() {
+                let timer;
+                // This is to restore the draft when the page loads and programatic click occurs
+                let pageloadclick = false;
+
+                // Bind click to the reply
+                $('.hsuforum-reply-link').on('click', function() {
+                    window.drafttext = '';
+
+                    let parent = $(this).closest('.hsuforum-post-wrapper');
+                    let postid = $(parent).attr('data-postid');
+
+                    fetchAutoSave(forumid, discussionid, postid, userid);
+
+                    setTimeout(function() {
+                        $(parent).find('#hiddenadvancededitoreditable').html(window.drafttext);
+                    }, 1000);
```

Comment:
Have you tested the timeout with browser throttling and in a big post? Trying to understand if the timeout is sufficient.

Reply:
Didn't test with throttling, but looking at that, should be able to pass the jQuery object as an additional param and then inject the HTML as part of the success function, that way we will never have to worry about race conditions

Reply:
So I think the reason this was done was because the node might not exist currently when the AJAX is called (because it is dynamically injected). I've made it a bit more robust by having a setInterval with 200ms that will run for 5 iterations (1s in total) or until it finds the injected content, and will then dispatch the AJAX request to restore the draft, and then inject